include/IFrameBuffer.h

```diff
@@ -50,12 +53,31 @@ class IRenderBuffer;
 class IFrameBuffer : public virtual IReferenceCounted
 {
     public:
+		//! Attaches given texture to given attachment point.
+		/** @param attachmentPoint Identifies attchment point.
+		@param tex Texture being attached.
+		@param mipMapLayer
+		@param layer
+		@returns Whether attachment operation went succesfully.
+		*/
         virtual bool attach(const E_FBO_ATTACHMENT_POINT &attachmenPoint, ITexture* tex, const uint32_t &mipMapLayer=0, const int32_t &layer=-1) = 0;
 
+		//! Attaches given render buffer to given attachment point.
+		/** @param attachmentPoint Identifies attchment point.
+		@param rbf Render buffer being attached.
+		@returns Whether attachment operation went succesfully.
+		*/
         virtual bool attach(const E_FBO_ATTACHMENT_POINT &attachmenPoint, IRenderBuffer* rbf) = 0;
 
+		//! Rebinds frame buffer object.
+		/** @returns whether FBO is bound or was successfully rebound.
+		*/
         virtual bool rebindRevalidate() = 0;
```

Comment:
This is a general function present in all our video namespace objects which reference/use resizeable resources such as textures and buffers.
ARB_buffer_storage and ARB_texture_storage remove the ability to respecify storage for OpenGL objects (essentially no GPU realloc) so to resize objects we need to destroy the old and create a new object, this means that the OpenGL handles (object IDs) change so the new object needs to be bound in place of the old (texture to an FBO, buffer to a VAO).
Whenever a bound object gets modified this way an atomic counter increases and this triggers the rebind* function to do more than just return `true`.

The driver always calls these functions internally before using the object with a rebind*() function.

However for the FrameBuffer you can call it manually after changing attachments as it will also call `glCheckFramebufferStatus` which will make the rebindRevalidate() function return false if the color formats you are trying to render to are invalid (like RGB9_E5) or if a certain combination of attachments is invalid (try specifying the DEPTH and STENCIL attachments to be different with any Nvidia GPU).

This function does nothing if no changes to bound objects have occurred.